swarm/src/handler.rs

```diff
@@ -210,6 +214,10 @@ pub enum ConnectionEvent<'a, IP: InboundUpgradeSend, OP: OutboundUpgradeSend, IO
     DialUpgradeError(DialUpgradeError<OOI, OP>),
     /// Informs the handler that upgrading an inbound substream to the given protocol has failed.
     ListenUpgradeError(ListenUpgradeError<IOI, IP>),
+    /// The local [`ConnectionHandler`] now supports a different set of protocols.
```

Comment:
```suggestion
    /// The local [`ConnectionHandler`] added or removed support for one or more protocols.
```

The comment sounds to me like the event includes the new set of protocols, whereas it actually only contains the difference.